dart/lib/src/open_location_code.dart

```diff
@@ -134,8 +134,8 @@ bool isValid(String code) {
     if (padMatch.length != 1) {
       return false;
     }
-    var matchLength = padMatch.first.group(0).length;
-    if (matchLength.isOdd || matchLength > separatorPosition - 2) {
+    var matchLength = padMatch.first.group(0)?.length;
```

Comment:
If `padMatch.first.group(0)` may be null, its length should probably not be asserted to be non-null in the next line.